### PR TITLE
Convert confusable violations to struct fields

### PR DESCRIPTION
**Pull request by sladyn98**
This PR adds converts confusable violations to struct fields.

### CODE REVIEW

File: src/checkers/ast.rs [hunk 1]
```diff
@@ -4472,7 +4472,7 @@ impl<'a> Checker<'a> {
                     let multiple = unused_imports.len() > 1;
                     for (full_name, range) in unused_imports {
                         let mut diagnostic = Diagnostic::new(
-                            violations::UnusedImport(full_name.to_string(), ignore_init, multiple),
+                            violations::UnusedImport{full_name.to_string(), ignore_init, multiple, name: todo!() },
```
Comment (charliermarsh) [inline src/checkers/ast.rs:4475]: Unfortunately this won't compile. Have you been able to set up a local development environment for Rust?

Comment (sladyn98) [reply src/checkers/ast.rs:4475]: Yeah I have actually but somehow when I run the tests it throws up 
``
Found 2806 error(s).
747 potentially fixable with the --fix option.
```

Comment (sladyn98) [reply src/checkers/ast.rs]: @charliermarsh Yeah fixed the compile errors.

File: src/checkers/ast.rs [hunk 2]
```diff
@@ -4472,7 +4472,7 @@ impl<'a> Checker<'a> {
                     let multiple = unused_imports.len() > 1;
                     for (full_name, range) in unused_imports {
                         let mut diagnostic = Diagnostic::new(
-                            violations::UnusedImport(full_name.to_string(), ignore_init, multiple),
+                            violations::UnusedImport{name:full_name.to_string(), ignore_init, multiple},
```
Comment (charliermarsh) [inline src/checkers/ast.rs:4475]: I'll go ahead and do it in this case, but for the future, you'll need to run `cargo +nightly fmt` to get the code in good shape for CI.

**Pull request merged**